Client/Frontend/Menu/Items/PagingMenuItemCollectionViewLayout.swift

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+class PagingMenuItemCollectionViewLayout: UICollectionViewLayout {
+
+    var maxNumberOfItemsPerPageRow: Int = 0
+    var lineSpacing: CGFloat = 0
+    var interitemSpacing: CGFloat = 0
+    var cellSize: CGSize = CGSize.zero
+
+    var contentSize = CGSize.zero
+    var layoutCellAttributes = [NSIndexPath: UICollectionViewLayoutAttributes]()
+
+    override func collectionViewContentSize() -> CGSize {
+        return contentSize
+    }
+
+    override func prepareLayout() {
+        guard let collectionView = self.collectionView,
+            let collectionViewDataSource = collectionView.dataSource else {
+                return super.prepareLayout()
```

Comment:
Nit: Not sure if it matters but usually you should always call super. Would it make sense to move this above and do a plain return here?